actor/src/main/resources/reference.conf

```diff
@@ -482,6 +482,10 @@ pekko {
         # Setting to "FIFO" to use queue like peeking mode which "poll" or "LIFO" to use stack
         # like peeking mode which "pop".
         task-peeking-mode = "FIFO"
+
+        # This config is new in Pekko v1.1.0 and only has an effect if you are running with JDK 9 and above.
+        # Read the documentation on `java.util.concurrent.ForkJoinPool` to find out more. Default in hex is 0x7fff.
+        maximum-pool-size = 32767
```

Comment:
These are huge too, will cause OOM on common 4core 8GB box

Reply:
please read the java.util.concurrent.ForkJoinPool code in Java 9+

this is the value hat it uses by default

Reply:
May be it makes sense to make to turn around the configuration for FJP and make it similar to how the default global FJP in JDK is configured:
- "parallelism" calculation could stay the same, be derived from runtime CPU count
- add a new parameter called something like `max-spare-threads` (similar to "java.util.concurrent.ForkJoinPool.common.maximumSpares" system property) with 256 as the default value (again the same as the default for the global FJP)
- the resulting maximum pool size will be calculated as parallelism + max-spare-threads

Additionally I think another parameter needs to be exposed - `reject-blocking-if-pool-saturated` or something similar which governs the `saturate` predicate in the new FJP constructor - the user should have a choice between failing on blocking task once the pool limit is reached or keep piling the tasks in memory.

Does it make sense?

Reply:
I'm not changing how the existing configs work - I worry far more about breaking behaviour for existing users than adding new features that only a small number of people want.

If we want new behaviour, we need new configs and to deprecate the old ones.

Please don't complicate this PR by adding extra requirements to it. Raise a new issue if you don't like the behaviour of the existing configs.

Reply:
Thank you! I see your point now about changing the existing behaviour. I just want to suggest that instead of exposing `maximum-pool-size` directly, it might be better to expose the spare thread count as JDK devs did for the global pool. The default value could be similar to the current default (effectively no limit at all).
Given that parallelism is calculated programmatically in the current Akka FJP configurator, combining it with a `maximum-pool-size` when the values are close might be inconvenient. Usually when you look at it, you think in the terms of "how many parallel blocking operations I can afford, additionally to the normal parallelism level".

I can contribute exposing `saturate` config in a separate PR later if this one goes through.

Reply:
The virtual thread is setting to `256` by default if the `maxPoolSizeValue` is not set, but yes, we can not change it and user can set to a limit match their systeam.

Reply:
I'm not sure what these comments mean. If someone wants changes, can they provide actual code. I can interpret code more easily than abstract comments.

I have read and re-read the ForkJoinPool code and if we use the old Java constructor (as opposed to the new one added in Java 9) - that the old constructor calls the new constructor and passes the value of MAX_CAP for the maxPoolSizeValue. MAX_CAP=0x7fff which is 32767.

I will make changes if I am wrong but please use code and code links to show me where I am going wrong.

Reply:
1. I mean your current code is right, 
2. we can not change it to a smaller number,eg 256
3. user can change it  if they like.